### PR TITLE
OLS-499:   add data collection settings in OLSConfig CRD.

### PR DESCRIPTION
**Pull request by raptorsun**
## Description

 add data collection settings in OLSConfig CRD.

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes # [OLS-499](https://issues.redhat.com//browse/OLS-499)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/ols_app_server_assets_test.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ var _ = Describe("App server assets", func() {
 			}))
 			Expect(dep.Spec.Template.Spec.Containers[1].Image).To(Equal(rOptions.LightspeedServiceImage))
 			Expect(dep.Spec.Template.Spec.Containers[1].Name).To(Equal("lightspeed-service-user-data-collector"))
-			Expect(dep.Spec.Template.Spec.Containers[0].Resources).ToNot(BeNil())
+			Expect(dep.Spec.Template.Spec.Containers[1].Resources).ToNot(BeNil())
```
Comment (bparees) [inline internal/controller/ols_app_server_assets_test.go:248]: shouldn't we expect both containers(apiserver and collector) to have non-nil resources?

**Review by bparees — COMMENTED** [review]
mostly lgtm, one question

File: internal/controller/ols_app_server_assets_test.go (hunk 1 above)
Comment (raptorsun) [reply internal/controller/ols_app_server_assets_test.go:248]: The application container has been checked already at [line 210](https://github.com/openshift/lightspeed-operator/pull/169/files#diff-3125771f1a7fef7f30b28ad2a5195cbf160890fb0e79187474ebe7dab08fe60aR210). This group of checks should be for data collector container.

Comment (bparees) [reply internal/controller/ols_app_server_assets_test.go:248]: ah, got it.

**Pull request merged**